pdns/dnsdistdist/dnsdist-healthchecks.cc

```diff
@@ -471,6 +483,7 @@ void handleQueuedHealthChecks(FDMultiplexer& mplexer, bool initial)
           infolog("Timeout while waiting for the health check response (ID %d) from backend %s", data->d_queryID, data->d_ds->getNameWithAddr());
         }
 
+        ++data->d_ds->d_healthCheckMetrics.d_timeOuts;
         data->d_ds->submitHealthCheckResult(initial, false);
       }
       catch (const std::exception& e) {
```

Comment:
No increment of any counter here, some a few other cases in `handleQueuedHealthChecks()`. I'm not  sure if these cases are already handled by the callback functions or not. At least a comment would be nice.

Reply:
I added a couple comments